x/tally/types/filters.go

```diff
@@ -215,51 +223,61 @@ func detectOutliersSignedInteger[T constraints.Integer](dataList []any, maxSigma
 	return outliers, true
 }
 
-func detectOutliersUnsignedInt[T constraints.Unsigned](dataList []any, maxSigma Sigma, errors []bool, replicationFactor uint16) ([]bool, bool) {
-	nums := make([]T, 0, len(dataList))
+func detectOutliersBigInt(dataList []string, _ Sigma, errors []bool, replicationFactor uint16) ([]bool, bool) {
+	sum := new(big.Int)
+	nums := make([]*big.Int, 0, len(dataList))
 	corruptQueue := make([]int, 0, len(dataList)) // queue of corrupt indices in dataList
 	for i, data := range dataList {
-		if data == nil {
+		if data == "" {
 			errors[i] = true
 			corruptQueue = append(corruptQueue, i)
 			continue
 		}
-		var num T
-		num1, ok := data.(int64)
-		if !ok {
-			jsonNum, ok := data.(json.Number)
-			if !ok {
-				errors[i] = true
-				corruptQueue = append(corruptQueue, i)
-				continue
-			}
 
-			num2, err := strconv.ParseUint(jsonNum.String(), 10, 64)
-			if err != nil {
-				errors[i] = true
-				corruptQueue = append(corruptQueue, i)
-				continue
-			}
-			num = T(num2)
-		} else {
-			num = T(num1)
+		num := new(big.Int)
+		_, ok := num.SetString(data, 10)
+		if !ok {
+			errors[i] = true
+			corruptQueue = append(corruptQueue, i)
+			continue
+		}
+		if num.BitLen() > 256 {
+			errors[i] = true
+			corruptQueue = append(corruptQueue, i)
+			continue
 		}
 		nums = append(nums, num)
+		sum.Add(sum, num)
 	}
 
 	// Construct outliers list.
 	outliers := make([]bool, len(dataList))
 	if len(nums) == 0 {
 		return outliers, false
 	}
-	median := findMedian(nums)
+
+	// Find sample standard deviation.
+	n := big.NewInt(int64(len(nums)))
+	mean := sum.Div(sum, n)
+
+	sumSquaredDiff := new(big.Int)
+	for _, num := range nums {
+		diff := new(big.Int).Sub(num, mean)
+		diff.Mul(diff, diff)
+		sumSquaredDiff.Add(sumSquaredDiff, diff)
+	}
+
+	variance := new(big.Int).Div(sumSquaredDiff, n.Sub(n, big.NewInt(1)))
+	stdDev := new(big.Int).Sqrt(variance)
+
+	// Fill out the outliers list.
 	var numsInd, nonOutlierCount int
 	for i := range outliers {
 		if len(corruptQueue) > 0 && i == corruptQueue[0] {
 			outliers[i] = true
 			corruptQueue = corruptQueue[1:]
 		} else {
-			if median.IsWithinSigma(nums[numsInd], maxSigma) {
+			if isWithinStdDev(nums[numsInd], mean, stdDev) {
```

Comment:
We should use sigma in here.

Instead of `stdDev` we could use something like this:

```
maxDev := new(big.Int).Mul(stdDev, big.NewInt(int64(maxSigma)))
```

Then it should be like:
```suggestion
			if isWithinStdDev(nums[numsInd], mean, maxDev) {
```

Reply:
Should we not first convert the max sigma to a number with 6 decimals?

Reply:
fixed to 
```go
maxDev := new(big.Int).Mul(stdDev, new(big.Int).SetUint64(maxSigma.WholeNumber()))
```

Reply:
I suppose it will be good to support fractional max sigma here? I think we can do some big rational math in `isWithinStdDev`

Reply:
I also renamed max sigma to sigma multiplier.